demo/js/application.js

```diff
@@ -4,52 +4,165 @@ let readline = require('readline-sync');
+        this._printer.basicMessage('D - Delete a book of the library.');
+        this._printer.basicMessage('exit - Leave de application');
 
         let response = '';
         response = readline.question('');
 
         switch(response){
         case 'L':
-            library.listBookOfLibrary();
+            this._library.listBookOfLibrary();
             break;
         case 'A':
-            printer.basicMessage('A');
+            this.addBookChoice();
             break;
         case 'U':
-            printer.basicMessage('U');
+            this.updateBookChoice();
             break;
         case 'D':
-            printer.basicMessage('D');
+            this.deleteBookChoice();
             break;
         case 'exit':
             process.exit(0);
             break;
         default:
-            printer.warningMessage('your choice is not an option');
+            this._printer.warningMessage('your choice is not an option');
             return false;
         }
         return true;
     }
+
+
+    addBookChoice(){
+        let isbn = '';
+        let title = '';
+        let description = '';
+
+        this._printer.paragrapheStartMessage('Please enter the following book\'s informations');
+
+        this._printer.paragrapheStartMessage('Isbn : ');
+        isbn = readline.question('');
+
+        this._printer.paragrapheStartMessage('Title : ');
+        title = readline.question('');
+
+        this._printer.paragrapheStartMessage('Description : ');
+        description = readline.question('');
+
+        if (!this._library.doesIsbnExists(isbn)){
+            this._library.addBook(isbn, title, description);
+        }
+        else{
+            this._printer.warningMessage('A book with this isbn is already in the library');
+        }
+    }
+
+    updateBookChoice(){
+        let response = '';
+        let isbn = '';
+        let title = '';
+        let description = '';
+        let id = '';
+        let bookfound = null;
+
+        while(true){
+            while (bookfound === null){
+                this._printer.paragrapheStartMessage('Please pick the book\'s id or type return');
+
+                id = readline.question('');
+                if (id === 'return')
+                    return true;
+
+                bookfound = this._library.getBookById(id);
+
+                if (bookfound === null){
+                    this._printer.warningMessage('No book with this id exists! Try again.');
+                }
+            }
+
+            this._printer.paragrapheStartMessage('Please choose a letter to pick an option : ');
+            this._printer.basicMessage('I - Update a book\'s isbn.');
+            this._printer.basicMessage('T - Update a book\'s title.');
+            this._printer.basicMessage('D - Update a book\'s description.');
+            this._printer.basicMessage('return - Go back to main menu.');
+
+            response = readline.question('');
+
+            switch(response){
```

Comment:
Use a function map instead of a switch case, and define the functions out of the structure